lib/rubocop/cop/rspec/sort_metadata.rb

```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Sort RSpec metadata alphabetically.
+      #
+      # @example
+      #   # bad
+      #   describe 'Something', :b, :a
+      #   context 'Something', foo: 'bar', baz: true
+      #   it 'works', :b, :a, foo: 'bar', baz: true
+      #
+      #   # good
+      #   describe 'Something', :a, :b
+      #   context 'Something', baz: true, foo: 'bar'
+      #   it 'works', :a, :b, baz: true, foo: 'bar'
+      #
+      class SortMetadata < Base
+        extend AutoCorrector
+        include RangeHelp
+
+        MSG = 'Sort metadata alphabetically.'
+
+        # @!method rspec_metadata(node)
+        def_node_matcher :rspec_metadata, <<~PATTERN
+          (block
+            (send
+              #rspec? {#Examples.all #ExampleGroups.all #SharedGroups.all #Hooks.all #Includes.all}
+              _ ${send str sym}* (hash $...)?)
+            ...)
+        PATTERN
+
+        # @!method rspec_configure(node)
+        def_node_matcher :rspec_configure, <<~PATTERN
+          (block (send #rspec? :configure) (args (arg $_)) ...)
+        PATTERN
+
+        # @!method metadata_in_block(node)
+        def_node_search :metadata_in_block, <<~PATTERN
+          (send (lvar %) #Hooks.all _ ${send str sym}* (hash $...)?)
+        PATTERN
+
+        def on_block(node)
+          rspec_configure(node) do |block_var|
+            metadata_in_block(node, block_var) do |symbols, pairs|
+              investigate(symbols, pairs.flatten)
+            end
+          end
+
+          rspec_metadata(node) do |symbols, pairs|
+            investigate(symbols, pairs.flatten)
+          end
+        end
+
+        alias on_numblock on_block
```

Comment:
@Darhazer I have no idea if this is correct. Tbh I think this message could be quite a bit clearer:

> lib/rubocop/cop/rspec/sort_metadata.rb:44:9: C: InternalAffairs/NumblockHandler: Define on_numblock to handle blocks with numbered arguments.